### ISSUE TITLE
 PLR0912: branches inside `with` statement not counted

### ISSUE THREAD

**Issue by Kakadus** [astral-sh/ruff#11313]
PLR0912(too-many-branches) does not count branches inside `with: ` blocks. With pylint the following is detected, while ruff does not detect this rule violation 
```py
from contextlib import suppress
def capital(country):
    with suppress(Exception):
        if country == "Australia":
            return "Canberra"
        elif country == "Brazil":
            return "Brasilia"
        elif country == "Canada":
            return "Ottawa"
        elif country == "England":
            return "London"
        elif country == "France":
            return "Paris"
        elif country == "Germany":
            return "Berlin"
        elif country == "Poland":
            return "Warsaw"
        elif country == "Romania":
            return "Bucharest"
        elif country == "Spain":
            return "Madrid"
        elif country == "Thailand":
            return "Bangkok"
        elif country == "Turkey":
            return "Ankara"
        elif country == "Undefined":
            return None
        else:
            return "Some"

```
Ruff version: 0.4.3
command: `ruff check test.py --select PLR0912 --isolated`

**Comment by charliermarsh** [comment]
Seems like a bug.

**Comment by dhruvmanila** [comment]
This seems like an easy fix which is to just include the `Stmt::With` in this function:

https://github.com/astral-sh/ruff/blob/868bbd4de6e193352d2d9a44be2f323a0887dd51/crates/ruff_linter/src/rules/pylint/rules/too_many_branches.rs#L88-L88

And then to add some test cases.
